src/utils/generator.js

```diff
@@ -146,9 +146,9 @@ export class Generator {
     });
   }
 
-  emitDefineProperty(object: ObjectValue, key: string, desc: Descriptor) {
+  emitDefineProperty(object: ObjectValue, key: string, desc: Descriptor, isDescChanged?: boolean) {
     if (object.refuseSerialization) return;
-    if (desc.enumerable && desc.configurable && desc.writable && desc.value) {
+    if (desc.enumerable && desc.configurable && desc.writable && desc.value && !isDescChanged) {
```

Comment:
There are several places in the codebase that check for "desc.enumerable && desc.configurable && desc.writable". Can you make into a helper function called "hasDefaultAttributeValues()"? Thanks.

Reply:
It seems there are only two occurrences of this in properties.js and one occurrence in generator.js, while the rests have different logic. It may be better keep current way since it is more explicit.